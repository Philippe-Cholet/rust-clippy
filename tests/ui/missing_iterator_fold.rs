#![warn(clippy::missing_iterator_fold)]

struct Countdown(u8);

impl Iterator for Countdown {
    type Item = u8;

    fn next(&mut self) -> Option<u8> {
        self.0 = self.0.checked_sub(1)?;
        Some(self.0)
    }
}
