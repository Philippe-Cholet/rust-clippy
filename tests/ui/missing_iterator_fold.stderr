error: you are implementing `Iterator` without specializing its `fold` method
  --> $DIR/missing_iterator_fold.rs:5:1
   |
LL | / impl Iterator for Countdown {
LL | |     type Item = u8;
LL | |
LL | |     fn next(&mut self) -> Option<u8> {
...  |
LL | |     }
LL | | }
   | |_^
   |
   = note: `-D clippy::missing-iterator-fold` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::missing_iterator_fold)]`

error: aborting due to 1 previous error

